web_session_allow_public/tests/test_ir_http.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import mock
+
+from openerp.tests.common import HttpCase
+
+from openerp.addons.base.ir.ir_http import ir_http
+
+
+MODULE_PATH = 'openerp.addons.base.ir.ir_http'
+
+
+class TestIrHttp(HttpCase):
+
+    @mock.patch('%s.request' % MODULE_PATH)
+    def test_public_session(self, request):
+        """ It should allow HTTP authentication on public user """
+        request.session.uid = False
+        with mock.patch.object(
```

Comment:
But the problem with current tests is that you are checking that the method raises the exception, but not checking if you actually can call XML/JSONRPC anonymously and still get access to public records. I mean, it's better than no tests, but still a little bit incomplete IMHO.